conan/tools/cmake/layout.py

```diff
@@ -2,6 +2,14 @@
 
 
 def cmake_layout(conanfile, generator=None, src_folder="."):
+    """
+
+    :param conanfile: The current recipe object. Always use ``self``.
+    :param generator: If specified, it will be used as the CMake generator if no declared
+                      ``tools.cmake.cmaketoolchain:generator``.
```

Comment:
```suggestion
    :param generator: Allow defining the CMake generator. In most cases it doesn't need to be passed, as it will get the value from the configuration              ``tools.cmake.cmaketoolchain:generator``, or it will automatically deduce the generator from the ``settings``
```